### PR TITLE
[KYUUBI #1458] Delta lake table columns won't show up in DBeaver.

### PR DESCRIPTION
**Pull request by mmuru**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/shim/CatalogShim_v2_4.scala [hunk 1]
```diff
@@ -125,7 +126,12 @@ class CatalogShim_v2_4 extends SparkCatalogShim {
     databases.flatMap { db =>
       val identifiers = catalog.listTables(db, tablePattern, includeLocalTempViews = true)
       catalog.getTablesByName(identifiers).flatMap { t =>
-        t.schema.zipWithIndex.filter(f => columnPattern.matcher(f._1.name).matches())
+        var tableSchema: StructType = null;
+        tableSchema = t.schema
+        if (t.provider.getOrElse("") == "delta") {
+          tableSchema = spark.table(t.identifier.table).schema
+        }
+        tableSchema.zipWithIndex.filter(f => columnPattern.matcher(f._1.name).matches())
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/shim/CatalogShim_v2_4.scala:134]: How about

``` scala
val tableSchema = if (t.provider.getOrElse("") == "delta") {
  spark.table(t.identifier.table).schema
} else {
  t.schema
}
```

BTW, is 'delta' here case-sensitive?

Comment (mmuru) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/shim/CatalogShim_v2_4.scala:134]: Yes, I changed it.  No, 'delta' is always lower case so I added equalsIgnoreCase.

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
please run `dev/reformat` https://github.com/apache/incubator-kyuubi/blob/master/dev/reformat to fix style issue

**Comment by mmuru** [comment]
@yaooqinn: Thanks for your review feedback.

**Review by yaooqinn — APPROVED** [review]

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**